hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystem.java

```diff
@@ -795,13 +796,15 @@ public void testListStatusOnKeyNameContainDelimiter() throws Exception {
    *
    * @throws IOException DB failure
    */
-  protected void deleteRootDir() throws IOException, InterruptedException {
+  protected void deleteRootDir()
+      throws IOException, InterruptedException, TimeoutException {
     FileStatus[] fileStatuses = fs.listStatus(ROOT);
 
     if (fileStatuses == null) {
       return;
     }
     deleteRootRecursively(fileStatuses);
+    Thread.sleep(500);
```

Comment:
How sleep for 500ms solves the problem?
And Who throws TimeoutException in what case make it flaky ?

Reply:
+1

I don't get either why `TimeoutException` is added [here](https://github.com/apache/ozone/pull/5217/files#diff-0a2a989278da5b28bd0346a9db79c98ca319ff238eb150af17e2686afabfc1e5R800). May be you forgot to remove it while reverting it.

Reply:
Since now GenericTestUtils is being used, I have not removed TimeoutException. This change is not directly related, however many other tests are getting impacted because of deleteRootDIr cleanup, so this is important as part of this test as well.